src/main/java/core/basesyntax/service/date/DateConverterImpl.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.service.date;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DateConverterImpl implements DateConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> rawData) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        if (rawData == null || rawData.isEmpty()) {
+            return transactions;
```

Comment:
Returning an empty list when the input data is null or empty may hide potential errors in the calling code. It might be better to throw an IllegalArgumentException in this case, to indicate that the method was called with invalid input.